src/mds/Beacon.cc

```diff
@@ -173,7 +177,18 @@ void Beacon::send_and_wait(const double duration)
   while (!seq_stamp.empty() && seq_stamp.begin()->first <= awaiting_seq) {
     auto now = clock::now();
     auto s = duration*.95-std::chrono::duration<double>(now-start).count();
-    if (s < 0) break;
+    if (s < 0) {
```

Comment:
What do you think about a timer task that is responsible for identifying events that are probably not working as expected (such as missing beacon, heartbeats, etc..)? In general anything that is expecting a reply within a certain interval can be monitored by such a task.

Reply:
Seems good idea. Will check that.